lib/utils/findMediaFeatureNames.mjs

```diff
@@ -115,28 +115,41 @@ function topLevelTokenNodes(node) {
 	/** @type {Array<import('@csstools/css-tokenizer').CSSToken>} */
 	const relevantTokens = [];
 
-	// To consume the next token if it is a scss variable
-	let lastWasDollarSign = false;
-
-	components.forEach((component) => {
-		// Only preserve top level tokens (idents, delims, ...)
-		// Discard all blocks, functions, ...
-		if (component && isTokenNode(component)) {
-			if (component.value[0] === TokenType.Delim && component.value[4].value === '$') {
-				lastWasDollarSign = true;
-
-				return;
-			}
+	// Only preserve top level tokens (idents, delims, ...)
+	// Discard all blocks, functions, ...
+	components.filter(isTokenNode).forEach((component, index, tokenNodes) => {
+		const [tokenType, , , ,] = component.value;
+
+		// E.g. $var
+		if (isScssVariableDelim(component)) {
+			return;
+		}
 
-			if (lastWasDollarSign) {
-				lastWasDollarSign = false;
+		if (tokenType === TokenType.Ident && isScssVariableDelim(tokenNodes[index - 1])) {
+			return;
+		}
 
-				return;
-			}
+		// E.g. namespace.$var
+		if (tokenType === TokenType.Ident && isScssNamespaceDelim(tokenNodes[index + 1])) {
+			return;
+		}
 
-			relevantTokens.push(component.value);
+		if (isScssNamespaceDelim(component)) {
+			return;
 		}
```

Comment:
[suggestion] Thank you for refactoring. In addition, can you align the use of `isScssVariableDelim` and `isScssNamespaceDelim`?

```js
// E.g. $var
if (isScssVariableDelim(component)) {

if (tokenType === TokenType.Ident && isScssVariableDelim(tokenNodes[index - 1])) {

// E.g. namespace.$var
if (isScssNamespaceDelim(component)) {

if (tokenType === TokenType.Ident && isScssNamespaceDelim(tokenNodes[index + 1])) {
```